services/121-service/src/registration/mappers/registration-views.mapper.ts

```diff
@@ -0,0 +1,116 @@
+import { RegistrationViewEntity } from '@121-service/src/registration/registration-view.entity';
+
+type RegistrationViewWithoutData = Omit<RegistrationViewEntity, 'data'>;
+
+export class RegistrationViewsMapper {
+  static mapRootRegistration({
+    registration,
+    select,
+    hasPersonalReadPermission,
+  }: {
+    registration: RegistrationViewEntity;
+    select?: string[];
+    hasPersonalReadPermission?: boolean;
+  }): RegistrationViewWithoutData {
+    let mappedRegistration = omit(registration, 'data');
+
+    if (select && select.length > 0) {
+      mappedRegistration = new RegistrationViewEntity();
+      for (const selectKey of select) {
+        if (selectKey !== 'data' && registration[selectKey] !== undefined) {
+          mappedRegistration[selectKey] = registration[selectKey];
+        }
+      }
+    }
+
+    if (!hasPersonalReadPermission) {
+      delete mappedRegistration.phoneNumber;
+    }
+
+    return mappedRegistration;
+  }
+
+  static mapRegistrationData(
+    registrationDataArray: RegistrationAttributeDataEntity[],
+    mappedRegistration: RegistrationViewWithoutData,
+    registrationDataInfoArray: RegistrationDataInfo[],
+  ) {
+    if (!registrationDataInfoArray || registrationDataInfoArray.length < 1) {
+      return mappedRegistration;
+    }
+
+    const findRelation = (
+      dataRelation: RegistrationDataRelation,
+      data: RegistrationAttributeDataEntity,
+    ): boolean => {
+      const propertiesToCheck = ['programRegistrationAttributeId'];
+      for (const property of propertiesToCheck) {
+        if (
+          dataRelation[property] === data[property] &&
+          data[property] !== null
+        ) {
+          return true;
+        }
+      }
+      return false;
+    };
+
+    for (const dataRelation of registrationDataInfoArray) {
+      const registrationData = registrationDataArray.find((x) =>
+        findRelation(dataRelation.relation, x),
+      );
+      if (registrationData) {
+        mappedRegistration[dataRelation.name] = registrationData.value;
+      } else {
+        mappedRegistration[dataRelation.name] = null;
+      }
+    }
+
+    return mappedRegistration;
+  }
+
+  static mapRegistrationName<T extends RegistrationViewWithoutData>({
+    registration,
+    select,
+    orignalSelect,
+    fullnameNamingConvention,
+  }: {
+    registration: T;
+    select?: string[];
+    orignalSelect: string[];
+    fullnameNamingConvention: string[];
+  }): T & { name: string } {
+    const name = this.getName(registration, fullnameNamingConvention);
+    if (select && select.includes('name')) {
+      const differenceOrignalSelect = select.filter(
+        (x) => !orignalSelect.includes(x),
+      );
+      for (const key of differenceOrignalSelect) {
+        delete registration[key];
+      }
+    }
+    return {
+      ...registration,
+      name,
+    };
+  }
+
+  static getName(
```

Comment:
should this - in line with this being a mapper - be something like 'mapRegistrationNamestoName'? (or something better)